lib/media/drm_engine.js

```diff
@@ -1942,17 +1944,27 @@ shaka.media.DrmEngine = class {
       }
     }
 
-    // Chromecast has a variant of PlayReady that uses a different key
-    // system ID.  Since manifest parsers convert the standard PlayReady
-    // UUID to the standard PlayReady key system ID, here we will switch
-    // to the Chromecast version if we are running on that platform.
-    // Note that this must come after fillInDrmInfoDefaults_, since the
-    // player config uses the standard PlayReady ID for license server
-    // configuration.
-    if (window.cast && window.cast.__platform__) {
-      if (drmInfo.keySystem == 'com.microsoft.playready') {
+    if (drmInfo.keySystem == 'com.microsoft.playready') {
+      // Chromecast has a variant of PlayReady that uses a different key
+      // system ID.  Since manifest parsers convert the standard PlayReady
+      // UUID to the standard PlayReady key system ID, here we will switch
+      // to the Chromecast version if we are running on that platform.
+      // Note that this must come after fillInDrmInfoDefaults_, since the
+      // player config uses the standard PlayReady ID for license server
+      // configuration.
+      if (window.cast && window.cast.__platform__) {
         drmInfo.keySystem = 'com.chromecast.playready';
```

Comment:
❓ Does `com.chromecast.playready` support robustness and if so should i invert the conditions so that playready keySystem on cast platform should **always** be `com.chromecast.playready` ?